pkg/config/mesh/kubemesh/watcher.go

```diff
@@ -18,112 +18,50 @@ import (
 	"istio.io/istio/pkg/kube"
-	"istio.io/istio/pkg/kube/watcher/configmapwatcher"
-	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/kube/kclient"
+	"istio.io/istio/pkg/kube/krt"
+	"istio.io/istio/pkg/ptr"
 )
 
-// NewConfigMapWatcher creates a new Watcher for changes to the given ConfigMap.
-func NewConfigMapWatcher(client kube.Client, namespace, name, key string, multiWatch bool, stop <-chan struct{}) *mesh.MultiWatcher {
-	w := mesh.NewMultiWatcher(mesh.DefaultMeshConfig())
-	c := configmapwatcher.NewController(client, namespace, name, func(cm *v1.ConfigMap) {
-		meshNetworks, err := ReadNetworksConfigMap(cm, "meshNetworks")
-		if err != nil {
-			// Keep the last known config in case there's a misconfiguration issue.
-			log.Errorf("failed to read meshNetworks config from ConfigMap: %v", err)
-			return
-		}
-		if meshNetworks != nil {
-			w.SetNetworks(meshNetworks)
-		}
-		if multiWatch {
-			meshConfig := meshConfigMapData(cm, key)
-			w.HandleMeshConfigData(meshConfig)
-			return
-		}
-		// Original behavior - just per-revision config
-		meshConfig, err := ReadConfigMap(cm, key)
-		if err != nil {
-			// Keep the last known config in case there's a misconfiguration issue.
-			log.Errorf("failed to read mesh config from ConfigMap: %v", err)
-			return
-		}
-		w.HandleMeshConfig(meshConfig)
-	})
-
-	go c.Run(stop)
-
-	// Ensure the ConfigMap is initially loaded if present.
-	if !client.WaitForCacheSync("configmap watcher", stop, c.HasSynced) {
-		log.Error("failed to wait for cache sync")
-	}
-	return w
-}
+const (
+	MeshConfigKey   = "mesh"
+	MeshNetworksKey = "meshNetworks"
+)
 
-func AddUserMeshConfig(client kube.Client, watcher mesh.Watcher, namespace, key, userMeshConfig string, stop <-chan struct{}) {
-	c := configmapwatcher.NewController(client, namespace, userMeshConfig, func(cm *v1.ConfigMap) {
-		meshConfig := meshConfigMapData(cm, key)
-		watcher.HandleUserMeshConfig(meshConfig)
+func NewConfigMapSource(client kube.Client, namespace, name, key string, opts krt.OptionsBuilder) meshwatcher.MeshConfigSource {
+	clt := kclient.NewFiltered[*v1.ConfigMap](client, kclient.Filter{
+		Namespace:     namespace,
+		FieldSelector: fields.OneTermEqualSelector(metav1.ObjectNameField, name).String(),
 	})
-
-	go c.Run(stop)
-	if !client.WaitForCacheSync("user mesh config", stop, c.HasSynced) {
-		log.Error("failed to wait for cache sync")
-	}
-}
-
-func meshConfigMapData(cm *v1.ConfigMap, key string) string {
-	if cm == nil {
-		return ""
-	}
-
-	cfgYaml, exists := cm.Data[key]
-	if !exists {
-		return ""
-	}
-
-	return cfgYaml
+	cms := krt.WrapClient(clt, opts.WithName("ConfigMap_"+name)...)
+
+	cmKey := types.NamespacedName{Namespace: namespace, Name: name}.String()
+
+	// Start informer immediately instead of with the rest. This is because we use configmapwatcher for
+	// single types (so its never shared), and for use cases where we need the results immediately
+	// during startup.
+	clt.Start(opts.Stop())
```

Comment:
Why wait until after creating `cmKey` and `cms`to start the informer

Reply:
No reason